core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -2147,6 +2147,53 @@ public void testSensitiveWrittenWhenOptionNotSet() {
                 B2Json.toJsonOrThrowRuntime(secureContainer, options));
     }
 
+    private static class OmitNullTestClass {
+        @B2Json.optional(omitNull = true)
+        private final String omitNullString;
+
+        @B2Json.optional
+        private final String regularString;
+
+        @B2Json.optional(omitNull = true)
+        private final int omitNullInt;
```

Comment:
Would it be clearer to include a field of type `Integer`, which could be null?  (In addition to this field, of type `int`, which is also a good test.)

Reply:
should putting omitNull on a non-nullable field be an error at the time we parse the annotations?

Reply:
Implemented both suggestions